editing/add-features-with-contingent-values/src/main/java/com/esri/samples/add_features_with_contingent_values/AddFeaturesWithContingentValuesController.java

```diff
@@ -0,0 +1,437 @@
+    if (protectionComboBox.getItems().size() == protectionFieldGroupValues.size()) {
+      protectionComboBox.getSelectionModel().selectFirst();
+    }
+  }
+
+  /**
+   * Handles interaction with the protection combo box. Gets the coded value chosen from the combo box and sets it as
+   * the feature's "Protection" attribute. Also sets the min and max value of the bufferSlider based on the contingent
+   * range value of the returned contingent range value.
+   */
+  @FXML
+  private void handleProtectionComboBox() {
+
+    if (protectionComboBox.getSelectionModel().getSelectedItem() != null) {
+      // put the value chosen from the protection combobox as the attribute for the new feature
+      newFeature.getAttributes().put("Protection", protectionComboBox.getSelectionModel().getSelectedItem().getCode());
+
+      // get contingent values for the buffer size field
+      ContingentValuesResult bufferContingentValues = geodatabaseFeatureTable.getContingentValues(newFeature,
+              "BufferSize");
+      // get contingent range values for the buffer size field group
+      ContingentRangeValue bufferRangeValue =
+              (ContingentRangeValue) bufferContingentValues.getContingentValuesByFieldGroup()
+                      .get("BufferSizeFieldGroup").get(0); // cast to required contingent value type
+
+      // get the minimum and maximum value from the buffer contingent range values
+      var minValue = (Integer) bufferRangeValue.getMinValue();
+      var maxValue = (Integer) bufferRangeValue.getMaxValue();
+
+      // set the min and max value of the buffer slider depending on the returned value of the contingent range value
+      bufferSlider.setMin(minValue);
+      bufferSlider.setMax(maxValue);
+      bufferSlider.setValue(minValue);
+      // set the "BufferSize" attribute to that of the value of the slider
+      newFeature.getAttributes().put("BufferSize", (int) bufferSlider.getValue());
+      // update label to show buffer size value
+      label.setText("Exclusion Area Buffer Size: " + (Math.round(bufferSlider.getValue())));
+    }
+  }
+
+  /**
+   * Handles interaction with the buffer slider. Gets the buffer slider value and sets is as the feature's "BufferSize"
+   * attribute.
+   */
+  @FXML
+  private void handleBufferSlider() {
+
+    // set the initial attribute and the text to the min of the contingent range value
+    newFeature.getAttributes().put("BufferSize", (int) Math.round(bufferSlider.getValue()));
+    label.setText("Exclusion Area Buffer Size: " + (Math.round(bufferSlider.getValue())));
+  }
+
+  /**
+   * Queries features in the geodatabase feature table, and creates a buffer for each feature based on its "BufferSize"
+   * attribute.
+   */
+  private void queryAndBufferFeatures() {
+
+    // get all the features that have buffer size greater than zero
+    var results = geodatabaseFeatureTable.queryFeaturesAsync(queryParameters);
```

Comment:
The queryParameters are only used in this method, so I'm wondering if it would be tidier to not have them as a global variable.  Instead they could be passed in to the queryAndBufferFeatures as a parameter.

Reply:
good spot: I think this was a hangover from a previous iteration!